### ISSUE TITLE
Ambiguous result of DockerSpringBootApplicationPlugin.getSpringApplicationMainClassName

### ISSUE THREAD

**Issue by fml2** [bmuschko/gradle-docker-plugin#778]
Hello,

thank you for the plugin!

I think, it has a potential issue. In the file **\src\main\groovy\com\bmuschko\gradle\docker\DockerSpringBootApplicationPlugin.groovy** there is the method **getSpringApplicationMainClassName**.

Its result depends on the order of the class dirs, because the method returns the first class with a 'SpringBootApplication' annotation.

To make the result not ambiguous, _all_ classes with this annotation should be found, then a check made that there is exactly one of them (failing the build when there is not).

**Comment by orzeh** [comment]
@fml2 sounds reasonable, however it seems to be a rare case? 
Nevertheless would you be interested in submitting PR?

**Comment by bmuschko** [comment]
I took that code from one of the Spring Boot libraries. I am not quite sure if they handle that case. You may want to have a look at it.

**Comment by cdancy** [comment]
I agree with @orzeh in that the request seems reasonable but I have to ask: how did you get yourself into this scenario in the first place? Multiple versions on the path or something?

**Comment by fml2** [comment]
I came to this issue by studying the plugin's code. I didn't know the spring boot plugin exists since I took over an old script that used java plugin. The created docker image contained all the librarires twice: once in the fat spring boot jar and once outside it. And then I wondered how the plugin figured out what class to use as the main class.

Long stroy short: I looked into the code, saw this place and thought that it's ambiguous. I didn't experience a real problem. But the code should be fixed nevertheless IMO.
